pkg/api/arm/resource.go

```diff
@@ -0,0 +1,174 @@
+	dst.Type = src.Type
+	if src.SystemData == nil {
+		dst.SystemData = nil
+	} else {
+		dst.SystemData = &SystemData{}
+		src.SystemData.Copy(dst.SystemData)
+	}
+}
+
+// TrackedResource represents a tracked ARM resource
+type TrackedResource struct {
+	Resource
+	Location string            `json:"location,omitempty"`
+	Tags     map[string]string `json:"tags,omitempty"`
+}
+
+func (src *TrackedResource) Copy(dst *TrackedResource) {
+	src.Resource.Copy(&dst.Resource)
+	dst.Location = src.Location
+	dst.Tags = maps.Clone(src.Tags)
+}
+
+// CreatedByType is the type of identity that created (or modified) the resource
+type CreatedByType int
+
+const (
+	CreatedByTypeApplication CreatedByType = iota
+	CreatedByTypeKey
+	CreatedByTypeManagedIdentity
+	CreatedByTypeUser
+
+	createdByTypeLength // private, must be last
+)
+
+func (v CreatedByType) String() string {
+	switch v {
+	case CreatedByTypeApplication:
+		return "Application"
+	case CreatedByTypeKey:
+		return "Key"
+	case CreatedByTypeManagedIdentity:
+		return "ManagedIdentity"
+	case CreatedByTypeUser:
+		return "User"
+	default:
+		return ""
+	}
+}
+
+func (v CreatedByType) MarshalText() (text []byte, err error) {
+	text = []byte(v.String())
+	if len(text) == 0 {
+		err = fmt.Errorf("Cannot marshal value %d", v)
+	}
+	return
+}
+
+func (v *CreatedByType) UnmarshalText(text []byte) error {
+	for i := range createdByTypeLength {
+		if i.String() == string(text) {
+			*v = i
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Cannot unmarshal '%s' to a %T enum value", string(text), *v)
+}
+
+// SystemData includes creation and modification metadata for resources
+// See https://eng.ms/docs/products/arm/api_contracts/resourcesystemdata
+type SystemData struct {
+	CreatedBy          string        `json:"createdBy,omitempty"`
+	CreatedByType      CreatedByType `json:"createdByType,omitempty"`
+	CreatedAt          *time.Time    `json:"createdAt,omitempty"`
+	LastModifiedBy     string        `json:"lastModifiedBy,omitempty"`
+	LastModifiedByType CreatedByType `json:"lastModifiedByType,omitempty"`
+	LastModifiedAt     *time.Time    `json:"lastModifiedAt,omitempty"`
+}
+
+func (src *SystemData) Copy(dst *SystemData) {
+	dst.CreatedBy = src.CreatedBy
+	dst.CreatedByType = src.CreatedByType
+	if src.CreatedAt == nil {
+		dst.CreatedAt = nil
+	} else {
+		t := time.Unix(src.CreatedAt.Unix(), 0)
+		dst.CreatedAt = &t
+	}
+	dst.LastModifiedBy = src.LastModifiedBy
+	dst.LastModifiedByType = src.LastModifiedByType
+	if dst.LastModifiedAt == nil {
+		dst.LastModifiedAt = nil
+	} else {
+		t := time.Unix(src.LastModifiedAt.Unix(), 0)
+		dst.LastModifiedAt = &t
+	}
+}
+
+// ProvisioningState represents the provisioning state of an ARM resource
+type ProvisioningState int
+
+const (
+	ProvisioningStateAccepted ProvisioningState = iota
+	ProvisioningStateCanceled
+	ProvisioningStateDeleting
+	ProvisioningStateFailed
+	ProvisioningStateProvisioning
+	ProvisioningStateSucceeded
+	ProvisioningStateUpdating
```

Comment:
TBD on any non-terminal states.

Reply:
Right.  Actually I was gonna reorder that list to put terminal states first, and I'll comment the rest as TBD for now.